go-controller/pkg/ovn/master.go

```diff
@@ -227,33 +124,11 @@ func (oc *Controller) upgradeOVNTopology(existingNodes *kapi.NodeList) error {
 // TODO: Verify that the cluster was not already called with a different global subnet
 //
 //	If true, then either quit or perform a complete reconfiguration of the cluster (recreate switches/routers with new subnet values)
-func (oc *Controller) StartClusterMaster() error {
+func (oc *DefaultNetworkController) StartClusterMaster() error {
 	klog.Infof("Starting cluster master")
 
 	metrics.RegisterMasterPerformance(oc.nbClient)
 	metrics.RegisterMasterFunctional()
```

Comment:
I would probably move `RegisterMasterPerformance` and `RegisterMasterFunctional` to the beginning of `NetworkControllerManager.configureMetrics`

While the functional metrics and some of the performance metrics might only be set by DefaultNetworkController, other performance metrics are rather global and not related to any specific controller (like MetricResourceAddLatency) for example. 

Registering the metrics itself could be considered a global thing to do and we probably can do it at an upper level without worrying about which controller plays with which metric.